### DISCUSSION TITLE
Welcome to Nimbus Discussions!

### DISCUSSION THREAD

**Discussion by DamianMac** [NimbusAPI/Nimbus#295] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by SaitoSoft** [comment]
HI Damian, The company I am working for are using using Nimbus currently with a Windows Service Bus, They wish to upgrade to Azure Service Bus however the large messaging support seems to have gone from the latest nuget package. I was wondering how is is this handled in the latest version (3.0) of Nimbus?

Many Thanks

Gary

**Reply by DamianMac** [reply]
Hey Gary

As it happens, I shipped large message support yesterday in our 4.x branch!
Nimbus has a habit of stalling a big, depending on what I'm personally doing. However over the past year I've been doing a lot with a client on Azure with .NET core. The develop branch is now on .NET 5 with lots of fixes and support for Azure Service Bus.

If you were to look at 4.2.0 in here https://myget.org/feed/nimbusapi/package/nuget/Nimbus there is large message support. 

Take a look at see if that works for you, if you need some help migrating let me know, perhaps we can turn it into some documentation.

**Reply by SaitoSoft** [reply]
Hi Damian,

Hope you are well. First of thanks for pointing me in the direction of the new packages. I have set them up and am able send messages to an Azure Queue with no problem however I have problem getting messages of the queue. I just wanted to clarify (via the documentation) that handlers cannot have any constructor dependencies? as I am getting the following error:

"No parameterless constructor defined for type 'ISG.ServiceBus.Handlers.Net5.UnitHoldingMessageHandler'"

"The DependencyResolverScope can only broker messages to handlers that have default constructors (i.e. ones with no parameters). If you'd like to use constructor injection on your handlers, have a look at the examples provided in the README about how to wire things up via an IoC container."

Using version 3.0 this seemed to be ok. Is there any way to use handlers in classes with constructor dependencies (using Autofac) with Nimbus or has this been eliminated in the new version?

**Reply by SaitoSoft** [reply]
Apologies I have attached 2 code files that work with V3. "UnitHoldingMessageHandler.txt" is the derived class and "GenericFileProcessorMessageHandler.txt" is the base class (no interface implementations). These are passed into the BUS component.

[UnitHoldingMessageHandler.txt](https://github.com/NimbusAPI/Nimbus/files/7287294/UnitHoldingMessageHandler.txt)
[GenericFileProcessorMessageHandler.txt](https://github.com/NimbusAPI/Nimbus/files/7287295/GenericFileProcessorMessageHandler.txt)

**Reply by DamianMac** [reply]
Hey Gary

You can definitely support constructor dependencies with Autofac, do you register Nimbus with autofac? Can you show me all your registration config?

**Reply by SaitoSoft** [reply]
Hey Damian

Great! thank you. I have attached 2 files "MainConfig" (this works fine starting Nimbus and sending messages) and "WorkerService" (which seems to work fine in polling the queues for messages)

[WorkerService.txt](https://github.com/NimbusAPI/Nimbus/files/7289763/WorkerService.txt)
[MainConfig.txt](https://github.com/NimbusAPI/Nimbus/files/7289764/MainConfig.txt)

**Reply by DamianMac** [reply]
Ahh looks like a gap in doco. 
In your `BuildBus` method you should add autofac using the BusBuilder extension.

```
.WithAutofacDefaults(ctx)
```

try that.
